[Gary Lee] iP

garyljj opened the pull request.

Commit:
Add Gradle support

Commit:
build.gradle: Update version to 8.29

garyljj committed:
Update gitignore

garyljj committed:
Implement echo

garyljj committed:
Checklist display numbered

garyljj committed:
Implement Task class

garyljj committed:
Implement done function

garyljj committed:
Implement Todo

garyljj committed:
Implement Deadline

garyljj committed:
Implement Event

garyljj committed:
Fix typo in Event

garyljj committed:
Fix typo in Duke

garyljj committed:
Update testcases

garyljj committed:
Implement DukeException to handle possible input error

garyljj committed:
Update testcases for level 5

garyljj committed:
Implement delete

garyljj committed:
Update testcases for level 6

garyljj committed:
Add documentation and cleanup

garyljj committed:
Fix indent and cleanup

garyljj committed:
Add import/exportData for read/write

garyljj committed:
Fix importData

garyljj committed:
Implement save, load

garyljj committed:
Update testcases for level 7

garyljj committed:
Use LocalDate for dates

garyljj committed:
Update testcase for level 8

garyljj committed:
Merge branch 'branch-Level-7'

garyljj committed:
Merge branch 'branch-Level-8'

garyljj committed:
Update javadocs

garyljj committed:
Add javadoc folder to gitignore

garyljj committed:
Add Date attribute to Task

garyljj committed:
Create Parser and DukeInputException

garyljj committed:
Created TaskList

garyljj committed:
Create Storage and checkImportFormat() in Parser

garyljj committed:
Create Ui

garyljj committed:
Implement Parser,Storage,TaskList,Ui into Duke

garyljj committed:
Update testcases for MoreOOP

garyljj committed:
Add help command and cleanup

garyljj committed:
Add JUnit test

garyljj committed:
Add javadocs

garyljj committed:
Cleanup code to follow coding standard

garyljj committed:
Implement search and update testcases

garyljj committed:
Merge branch 'branch-A-JavaDoc'

garyljj committed:
Merge branch 'branch-A-CodingStandard'

garyljj committed:
Merge branch 'branch-Level-9'

garyljj committed:
Implement save prompt on bye

garyljj committed:
Include missing javadocs and cleanup code

garyljj committed:
Implement delete confirmation prompt

garyljj committed:
Shift test files to test folder

garyljj committed:
Merge branch 'add-gradle-support' of https://github.com/garyljj/ip

garyljj committed:
Add gradle support

garyljj committed:
Create duke package and edit runtest

Edit runtest to work with the package

garyljj committed:
Add checkstyle and edit to follow standard

georgepwhuang commented on src/main/java/duke/util/Ui.java:
```diff
@@ -0,0 +1,194 @@
+    /**
+     * Displays greeting messages when Duke starts.
+     */
+    public void greetings() {
```
Names representing methods must be verbs.

georgepwhuang commented on src/main/java/duke/util/Ui.java:
```diff
@@ -0,0 +1,194 @@
+    /**
+     * Displays messages after TaskList has been saved.
+     */
+    public void saved() {
```
Names representing methods must be verbs.

deyixtan replied:
I agree, should it be converted into a verb form?

georgepwhuang commented on src/main/java/duke/util/Ui.java:
```diff
@@ -0,0 +1,194 @@
+    /**
+     * Displays messages after TaskList has been loaded.
+     */
+    public void loaded() {
```
Names representing methods must be verbs.

georgepwhuang commented on src/main/java/duke/util/Deadline.java:
```diff
@@ -0,0 +1,75 @@
+
+        return new Deadline(details[0], date);
+    }
+
```
Write descriptive header comments for all public classes/methods.

deyixtan replied:
I agree with @georgepwhuang. Perhaps you could include them in?

georgepwhuang commented on src/main/java/duke/util/Event.java:
```diff
@@ -0,0 +1,75 @@
+
+        return new Event(details[0], date);
+    }
+
```
Write descriptive header comments for all public classes/methods

georgepwhuang commented on src/main/java/duke/util/Todo.java:
```diff
@@ -0,0 +1,56 @@
+        }
+        return new Todo(description);
+    }
+
```
Write descriptive header comments for all public classes/methods

georgepwhuang reviewed (commented):
LGTM

deyixtan commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,190 @@
+
+    /**
+     * Constructor to create Duke object.
+     * @param filePath File path to save tasklist.
```
Should there be an empty line between description and parameter section?

deyixtan commented on src/main/java/duke/util/Deadline.java:
```diff
@@ -0,0 +1,75 @@
+
+    /**
+     * Factory method for creating deadline task.
+     * @param input Description of the task and its due date.
```
Should there be an empty line between description and parameter section?

deyixtan commented on src/main/java/duke/util/Event.java:
```diff
@@ -0,0 +1,75 @@
+
+    /**
+     * Factory method for creating Event task.
+     * @param input Description of the task and its date.
```
Should there be an empty line between description and parameter section?

deyixtan commented on src/main/java/duke/util/Parser.java:
```diff
@@ -0,0 +1,169 @@
+    /**
+     * Checks if input from user is valid.
+     * <br>Only accepts valid commands, followed by valid parameters.
+     * @param input Input from user.
```
Should there be an empty line between description and parameter section?

deyixtan commented on src/main/java/duke/util/Event.java:
```diff
@@ -0,0 +1,75 @@
+
+    /**
+     * Import data from standardised format.
+     */
```
Should `@param` and `@return` be included?

deyixtan commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,190 @@
+ * <br>search [keyword/date]
+ * <br>  - Display all task containing the following keyword.
+ * <br>  - If keyword is in a valid date format(YYYY-MM-DD), display all task on that date.
+ */
```
Is it too descriptive? Perhaps have it more concise?

deyixtan commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,190 @@
+        }
+    }
+
+    private void completeTask(String num) throws DukeInputException {
```
I noticed some methods did not have Javadoc comments. Perhaps, you could add them in?

deyixtan commented on src/main/java/duke/util/Deadline.java:
```diff
@@ -0,0 +1,75 @@
+    /**
+     * Export data into a standardised format.
+     */
+    @Override
```
I think the `@return` is missing?

deyixtan commented on src/main/java/duke/util/Deadline.java:
```diff
@@ -0,0 +1,75 @@
+    /**
+     * Import data from standardised format.
+     */
+    protected static Deadline importData(String[] args) {
```
Did you forget to include the `@param` and `@return`?

deyixtan commented on src/main/java/duke/util/Parser.java:
```diff
@@ -0,0 +1,169 @@
+    /**
+     * Checks if input from text file is valid.
+     * @param input Line from saved file.
+     * @throws DukeInputException If input is invalid.
```
I noticed some methods in multiple files that followed the same Javadoc format. Should there be an empty line between description and parameter section?

deyixtan commented on src/main/java/duke/util/Task.java:
```diff
@@ -0,0 +1,64 @@
+    /**
+     * Set task as completed.
+     */
+    public void completed() {
```
Perhaps can name it like `markComplete`?

deyixtan commented on src/main/java/duke/util/TaskList.java:
```diff
@@ -0,0 +1,144 @@
+     * Returns size of tasklist.
+     * @return Size of tasklist.
+     */
+    public int size() {
```
I think the method name should be in a verb form?

deyixtan reviewed (commented):
Except for some nits to fix, your code complies with the coding standard. LGTM.

garyljj committed:
Cleanup code from PR feedback

garyljj committed:
Implement skeleton of GUI

garyljj committed:
Convert TaskList's List to ObservableList

garyljj committed:
Update gitignore

garyljj committed:
Implement GUI

garyljj committed:
Merge branch 'branch-Level-10'

garyljj committed:
Add javadocs, Cleanup GUI

garyljj committed:
Implement varargs in Ui

garyljj committed:
Add more testcases

garyljj committed:
Beautify GUI

garyljj committed:
Add assertions

garyljj committed:
Refactor to improve code quality

garyljj committed:
Add more testcases and fix typo in Parser

garyljj committed:
Implement multiple args for done,delete

garyljj committed:
Merge pull request #1 from garyljj/branch-A-Assertions

Add assertions

garyljj committed:
Merge branch 'master' of https://github.com/garyljj/ip

garyljj committed:
Merge branch 'master' into branch-A-CodeQuality

garyljj committed:
Merge pull request #3 from garyljj/branch-A-CodeQuality

Fix code quality violations

garyljj committed:
Merge branch 'master' into branch-C-MassOps

garyljj committed:
Merge pull request #4 from garyljj/branch-C-MassOps

Add multiple args functionality for done,delete

garyljj committed:
Implement sort tasklist

garyljj committed:
Fix bug in tasklist

garyljj committed:
Merge branch 'C-Sort'

garyljj committed:
Abstract out getResponse and add missing javadocs

garyljj committed:
Add command enum and DukeTest

garyljj committed:
Implement high/low priority

garyljj committed:
Merge branch 'branch-C-Priority'

garyljj committed:
Implement textfield history and add gui package

garyljj committed:
Merge branch 'branch-History'

garyljj committed:
Fix typo in help message

garyljj committed:
Implement sample and clear

garyljj committed:
Merge branch 'branch-Sample'

garyljj committed:
Set theme jekyll-theme-cayman

garyljj committed:
Improve and finalize GUI and help text

garyljj committed:
Merge branch 'master' of https://github.com/garyljj/ip

garyljj committed:
Update README

garyljj committed:
Update README and remove theme

garyljj committed:
Update user guide

garyljj committed:
Update README

garyljj committed:
Shift Launcher out of package

garyljj committed:
Abstract CustomListCell out

garyljj committed:
Implement Pair class

garyljj committed:
Finalize Duke

Ensure variable names are following standard.
Rename some methods to verbs.
Abstracted out booleans.
Added minor comments.
Ensure javadocs for public methods.
Remove redundant checks in create task methods.

garyljj committed:
Fix bug in Parser

Disallow use of semicolon as input to avoid conflict with save file.
